.github/workflows/release.yml

```diff
@@ -1,23 +1,90 @@
 name: Publish Package to npmjs
 on:
   release:
-    types: [created]
+    types: [published]
 jobs:
-  build:
+  # Build package and increase version
+  build-version:
     permissions:
+      contents: write
       id-token: write
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PAT }}
+          ref: main
       # Setup .npmrc file to publish to npm
       - uses: actions/setup-node@v3
         with:
           node-version: '16.x'
           registry-url: 'https://registry.npmjs.org'
       # Ensure that the README is published with the package
       - run: rm -f packages/cli/README.md && cp README.md packages/cli
-      - run: npm ci --workspace packages/cli
-      - run: npm publish --provenance --workspace packages/cli
+      - run: npm ci
+      # Use release tag and update package version
+      - name: Set new version from release tag
+        run: |
+          git config user.name "Checkly Engineering"
+          git config user.email "engineering@checklyhq.com"
+          npm --no-git-tag-version version ${{ github.event.release.tag_name }} --workspace packages/cli
+          git add .
+      # Update the changelog with the release changes using pull-requests tags as categories
+      - name: Append changes to changelog
+        run: |
+          npx auto-changelog --latest-version ${{ github.event.release.tag_name }}
+          git add .
+      # Commit and push all changes to main
+      - name: Git push new package version
+        run: |
+          git commit -m "feat: release CLI ${{ github.event.release.tag_name }}"
+          git push origin main --follow-tags --force
+      - name: Output version change after push
```

Comment:
Why are we creating another tag here? We already created one using the Github UI anyways at this point, or?

Reply:
I'm gonna use the release tag without pushing back the `package.json` to the main branch. The version will be something like `0.0.1` in the `package.json`, or a un-synced version number comparing with the NPM version.